libtenzir/src/execution_node.cpp

```diff
@@ -686,7 +687,10 @@ struct exec_node_state {
           TENZIR_DEBUG("{} {} failed to get its demand fulfilled: {}", *self,
                        op->name(), err);
           issue_demand_inflight = false;
-          if (err != caf::sec::request_receiver_down) {
+          if (err
+              and err != caf::sec::request_receiver_down
+              // TODO: verify that this is fine.
+              and err != caf::exit_reason::remote_link_unreachable) {
```

Comment:
This is fine if my understanding is correct. Which of the two error codes you get is no longer deterministic for remote actors. That's a bit unfortunate, but whatever.